collector/exporter/otelarrowexporter/internal/arrow/stream.go

```diff
@@ -340,6 +341,14 @@ func (s *Stream) encodeAndSend(wri writeItem, hdrsBuf *bytes.Buffer, hdrsEnc *hp
 	ctx, span := s.tracer.Start(wri.parent, "otel_arrow_stream_send")
 	defer span.End()
 
+	var err error
```

Comment:
I think you might have the wrong `err` here vs. the one returned below, so this might not work. The pattern that I like here, is to use a named return value on line 340
```
func (s *Stream) encodeAndSend(wri writeItem, hdrsBuf *bytes.Buffer, hdrsEnc *hpack.Encoder) (retErr error) {
```
and then later, on line 415 when you `return err`, the `retErr` is assigned to before returning so your deferred function will see the proper error.

Reply:
Oh I see I didn't notice err got redefined below, thanks for catching that! Using your suggestion now